### PR TITLE
RFC: API Change: watchdog: wdt_feed error codes

### PR DESCRIPTION
**Pull request by pabigot**
Allow feed operations that would stall due to internal peripheral
delays to complete immediately with -EAGAIN indicating that the feed
did not occur as expected.

Supports #23115 and RFC #26708.

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]

**Review by tbursztyka — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@pabigot I updated the PR description to reflect the change to EAGAIN; you may wish to do the same for the commit log.

**Review by stephanosio — APPROVED** [review]

**Review by nzmichaelh — APPROVED** [review]
LGTM

**Comment by pabigot** [comment]
@nashif @carlescufi While I don't object to this being merged, it was supposed to follow a [stable change process](https://docs.zephyrproject.org/latest/development_process/api_lifecycle.html#introducing-incompatible-changes) where merge requires the approval of the subsystem maintainer.  It also was supposed to be discussed in detail in the API meeting; while I mentioned it was going to happen last meeting, but we had limited attendance and it hadn't yet been announced to the mailing list so we didn't discuss it in detail.

I had asked on Slack general who that maintainer is and got no response; I'd planned to follow up on that today.

**Pull request merged**
